src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -1277,37 +1277,36 @@ UpdateColocationGroupReplicationFactor(uint32 colocationId, int replicationFacto
 	/* we first search for colocation group by its colocation id */
 	pgDistColocation = heap_open(DistColocationRelationId(), RowExclusiveLock);
 	tupleDescriptor = RelationGetDescr(pgDistColocation);
-	ScanKeyInit(&scanKey[0], Anum_pg_dist_colocation_colocationid, BTEqualStrategyNumber,
-				F_OIDEQ, ObjectIdGetDatum(colocationId));
+	ScanKeyInit(&scanKey[0], Anum_pg_dist_colocation_distributioncolumntype,
+				BTEqualStrategyNumber,
+				F_OIDEQ, ObjectIdGetDatum(InvalidOid));
 
 	scanDescriptor = systable_beginscan(pgDistColocation,
-										DistColocationColocationidIndexId(), indexOK,
+										InvalidOid, indexOK,
 										NULL, scanKeyCount, scanKey);
 
 	heapTuple = systable_getnext(scanDescriptor);
-	if (!HeapTupleIsValid(heapTuple))
+	if (HeapTupleIsValid(heapTuple))
 	{
-		ereport(ERROR, (errcode(ERRCODE_UNDEFINED_OBJECT),
-						errmsg("could not find valid entry for colocation group "
-							   "%d", colocationId)));
-	}
-
-	/* after we find colocation group, we update it with new values */
-	memset(replace, false, sizeof(replace));
-	memset(isnull, false, sizeof(isnull));
-	memset(values, 0, sizeof(values));
+		/* after we find colocation group, we update it with new values */
```

Comment:
Minor: We could move declarations of `values/isnull/replace` and `newHeapTuple` inside the if statement.